### ISSUE TITLE
Suggestion to Enhance install_os_dependencies

### ISSUE THREAD

**Issue by burhan** [pydanny/cookiecutter-django#389]
On the latest Warewolf release; the dependencies are outdated:

```
Package libtiff4-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  libtiff5-dev:i386 libtiff5-dev

Package liblcms1-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libtiff4-dev' has no installation candidate
E: Package 'liblcms1-dev' has no installation candidate
```

Instead of patching the `requirements.apt` file, I would like to implement the following:
1. Keep two sets of requirements; one for the latest LTS and the other for the non-LTS release.
2. Implement a check in the script to detect which version of the release is running and install the appropriate apt packages.

In addition - and I am not sure what the demand is out there, but I would suggest including support for an `rpm`-based distribution as well, in the spirit of inclusion.

If this is something that sounds good, I can start working on PR.

**Comment by crdoconnor** [comment]
I actually bundled something like this in with the testing framework (currently a pending pull request: https://github.com/pydanny/cookiecutter-django/pull/382/files).

It uses this other project I created to detect and translate the apt name to rpm and arch and brew and installs them for you: https://github.com/unixpackage/unixpackage

I might add a libtiff-dev and get unixpackage to detect/trigger installation from that. I don't think cookiecutter/pillow is necessarily fussy about whether libtiff is version 4 or 5 (could be wrong about that though).
